includes/embeds/class-amp-wordpress-tv-embed-handler.php

```diff
@@ -0,0 +1,43 @@
+<?php
+/**
+ * Class AMP_WordPress_TV_Embed_Handler
+ *
+ * @package AMP
+ */
+
+/**
+ * Class AMP_WordPress_TV_Embed_Handler
+ */
+class AMP_WordPress_TV_Embed_Handler extends AMP_Base_Embed_Handler {
+
+	/**
+	 * Register embed.
+	 */
+	public function register_embed() {
+		add_filter( 'render_block', [ $this, 'filter_rendered_block' ], 0, 2 );
+	}
+
+	/**
+	 * Unregister embed.
+	 */
+	public function unregister_embed() {
+		remove_filter( 'render_block', [ $this, 'filter_rendered_block' ], 0 );
+	}
+
+	/**
+	 * Filters the content of a single block to make it AMP valid.
+	 *
+	 * @param string $block_content The block content about to be appended.
+	 * @param array  $block         The full block, including name and attributes.
+	 * @return string Filtered block content.
+	 */
+	public function filter_rendered_block( $block_content, $block ) {
+		if ( ! isset( $block['blockName'] ) || 'core-embed/wordpress-tv' !== $block['blockName'] ) {
+			return $block_content;
+		}
+
+		$modified_block_content = preg_replace( '#<script(?:\s.*?)?>.*?</script>#s', '', $block_content );
+
+		return null !== $modified_block_content ? $modified_block_content : $block_content;
```

Comment:
Just general defensive coding? Regex engines are changing between PHP versions, so this would ensure that, no matter what, we don't take the server down if something unexpected happens...

Reply:
OK, fair enough.